xmtp_db/src/encrypted_store/native.rs

```diff
@@ -124,6 +124,9 @@ impl RetryableError for NativeStorageError {
             Self::SqlCipherNotLoaded => true,
             Self::PoolNeedsConnection => true,
             Self::SqlCipherKeyIncorrect => false,
+            Self::DieselResult(result) => retryable!(result),
```

Comment:
This change snuck in. I think all of these should be safe to treat as retryable.